backend/contracts/ExpenseTracker.sol

```diff
@@ -17,75 +18,91 @@ contract ExpenseTracker {
     }
 
     struct Tracker {
-        uint total;
-        uint deleted;
-        string[] transactions;
+        uint256 total;
+        uint256 deleted;
+        mapping(uint256 => string) transactions;
     }
     mapping(address => mapping(string => Transaction)) private transactions;
     mapping(address => Tracker) private tracker;
     mapping(string => bool) private all_ids;
 
     function addExpense(
-        string memory _id,
         uint256 _amount,
         string memory _description,
         string memory _date,
         string memory _category
     ) public {
-        require(all_ids[_id] == false, "invalid id");
-        transactions[msg.sender][_id] = Transaction(
-            _id,
+        require(bytes(_category).length > 0, "category is required");
+        require(bytes(_date).length > 0, "date is required");
+        require(isValidDateFormat(_date), "invalid date format");
+        require(isValidDate(_date), "invalid date");
+
+        uint256 id = tracker[msg.sender].total;
+
+        Transaction memory expense = Transaction(

```

Comment:
The transaction should be stored in transactions mapping
mapping(address => mapping(string => Transaction)) private transactions;

Reply:
Actually I was saying that the Transaction of the user is assigned to expense which is local to function, but it should be added to the mapping.
transactions[msg.sender][id] = expense
Ans same thing for income.

Reply:
transactions[msg.sender][id] = expense
And same thing for income.     These to be added where can you specify the line no.??